### PR TITLE
Options refactor

### PR DESCRIPTION
**Pull request by JensWehner**
The beginning of the options refactor. 

- -p option now prints options to a file
- updated most calculators
- reworked orca option handling
- removed settings class
- added a lot of help text
- removed a lot of unnecessary options

closes #704

### CODE REVIEW

**Comment by JensWehner** [comment]
@votca-bot format

**Comment by JensWehner** [comment]
@votca-bot changelog: overhaul complete option handling

**Review by rubengerritsen — COMMENTED** [review]
Seems nice, readability is a lot better now in many places. It also *seems* like you fixed some issues we encountered while doing qmmm (you needed to set a lot of options before the polar region worked for example). I will look at the tools PR and then redo the LAMMPS tutorial with the new options to see if it really is easier.

File: include/votca/xtp/checkpointreader.h [hunk 1]
```diff
@@ -114,6 +115,22 @@ class CheckpointReader {
...
+    char pStr[128];
+    for (hsize_t i = 0; i < loc_.getNumObjs(); i++) {
+      memset(pStr, 0, 128);
+      loc_.getObjnameByIdx(i, pStr, 128);
+      try {
+        loc_.openGroup(pStr);
+        result.push_back(pStr);
+      } catch (H5::Exception&) {
+        ;
+      }
```
[the remark this answers is not in the document]
Comment (JensWehner) [reply include/votca/xtp/checkpointreader.h:129]: I did not find a `group.exists()` method in the hdf5 library, so I did it like this.

**Comment by rubengerritsen** [comment]
I have started with doing a simple `dftgwbse`. When I generate the options with

```bash
xtp_tools -p dftgwbse -o dftgwbse.xml
```
I get the following output

```bash
Writing options for calculator dftgwbse to dftgwbse.xml
nothing to be done - stopping here
```

Which reads like an error message (I know it isn't) which is confusing, because it had something to do and it even did it.

**Comment by rubengerritsen** [comment]
I don't know how much control there is over the ordering of the options in the `-p` generated xml files, but it is confusing that some options are so far from one another. Two examples the package name and package executable are separated by 5 lines and the functional and basisset options are not close either.

**Comment by rubengerritsen** [comment]
It also seems that there are still some options that could be hidden from users completely like the 

- scratch
- cleanup and
- temporary files (this is just an option that names a temporary file `temp`, that could be hard coded I think, the name of the file shouldn't matter it is temporary)

options

**Comment by rubengerritsen** [comment]
When you use orca to do the dft calculation it crashes now. It can't parse the functional.

To reproduce use the defaults and run a dftgwbse calculation on any molecule.

**Comment by JensWehner** [comment]
Thanks

**Comment by baumeier** [comment]
> It also seems that there are still some options that could be hidden from users completely like the
> 
> * scratch
> * cleanup and
> * temporary files (this is just an option that names a temporary file `temp`, that could be hard coded I think, the name of the file shouldn't matter it is temporary)
> 
> options

Regarding the `scratch` and `temp` options: if I am not mistaken, they were mainly used on cluster machines to move some IO intensive steps (like orca's scratch files) from network drives to fast local disks, no? Why do we have two?

**Comment by JensWehner** [comment]
What do you mean two? Scratch is the directory and temporary file is the filename used for the files. We could isolate all of this if we remove the `input,dft,parse` and turn it into `dft` and encapsulate the `qmpackages` much more.

**Comment by baumeier** [comment]
I wasn't sure anymore that `scratch` -> directory and `temp` -> filename.

**Review by rubengerritsen — APPROVED** [review]

**Comment by rubengerritsen** [comment]
@JensWehner can you fix the bad merge in your latest commit and then merge this in master?

**Comment by JensWehner** [comment]
@votca-bot format

**Comment by JensWehner** [comment]
I cannot merge it into master yet, because for that we have to fix the documentation.

**Comment by JensWehner** [comment]
@votca-bot format

**Comment by JensWehner** [comment]
@votca-bot format

File: share/xtp/xml/qmmm.xml [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
+    <max_iterations help="max iterations for qmmm scf loop" default="100" choices="int+" />
+    <map_file help="xml file with segment definition" default="votca_map.xml" />
+    <job_file help="name of jobfile to which jobs are written" default="qmmm_jobs.xml" />
+    <io_jobfile help="options to write a jobfile">
+      <states help="states to write jobs for and which to parse from jobfile" default="n e h" />
+      <segments help="segments to write jobs for and which to parse from jobfile" default="all"/>
+      <use_gs_for_ex help="If true uses the ground state geometry for excited states" default="false"/>
+    </io_jobfile>
+    <regions list="" help="Definitions of all regions inside qmmmm" default="REQUIRED">
+      <qmregion default="OPTIONAL" link="region.xml" help="qmregion with dft and optionally gwbse">
```
Comment (junghans) [inline share/xtp/xml/qmmm.xml:14]: @JensWehner you are referring to `region.xml` here, but didn't added that file, hence the documentation build fails.

Comment (JensWehner) [reply share/xtp/xml/qmmm.xml:14]: I think I fixed this now.

**Pull request merged**